metrics/telemetry.go

```diff
@@ -91,13 +98,17 @@ func GaugeVec(name string, labels []string) GaugeVecMeter {
 	return metrics.GetOrCreateGaugeVecMeter(name, labels)
 }
 
+var lazyLoadLock sync.Mutex
+
 // LazyLoad allows to defer the instantiation of the metric while allowing its definition. More clearly:
 // - it allow metrics to be defined and used package wide (using var)
 // - it avoid metrics definition to determine the singleton to use (noop vs prometheus)
 func LazyLoad[T any](f func() T) func() T {
 	var result T
 	var loaded bool
 	return func() T {
+		lazyLoadLock.Lock()
+		defer lazyLoadLock.Unlock()
```

Comment:
Not getting this one. Any examples?

Reply:
The websocket duration metric was causing issues when running:

```bash
make test-coverage
```

It seems there was a race condition where the below would be executed twice:

```golang
if !loaded {
	result = f()
	loaded = true
}
```

See this GHA: https://github.com/vechain/thor/actions/runs/12027304858/job/33528077050

Reply:
@libotony I was ablo to do:

```go
once.Do(func() {
	result = f()
})
```